src/files/infrastructure/mappers/JSFileMapper.ts

```diff
@@ -0,0 +1,170 @@
+      this.toFileEmbargo(jsFile.embargo),
+      this.toFileTabularData(),
+      this.toFileDescription(jsFile.description)
+    )
+  }
+
+  static toFileUserPermissions(
+    jsFileId: number,
+    jsFileUserPermissions: {
+      canDownloadFile: boolean
+      canEditOwnerDataset: boolean
+    }
+  ): FileUserPermissions {
+    return {
+      fileId: jsFileId,
+      canDownloadFile: jsFileUserPermissions.canDownloadFile,
+      canEditDataset: jsFileUserPermissions.canEditOwnerDataset
+    }
+  }
+
+  static toFileId(jsFileId: number): number {
+    return jsFileId
+  }
+
+  static toFileVersion(
+    jsVersion: number,
+    datasetVersion: DatasetVersion,
+    jsPublicationDate?: Date
+  ): FileVersion {
+    const fileVersion = { number: jsVersion, publishingStatus: FilePublishingStatus.DRAFT }
+
+    if (datasetVersion.publishingStatus === DatasetPublishingStatus.DEACCESSIONED) {
+      fileVersion.publishingStatus = FilePublishingStatus.DEACCESSIONED
+    }
+
+    if (jsPublicationDate) {
+      fileVersion.publishingStatus = FilePublishingStatus.RELEASED
+    }
+
+    return fileVersion
+  }
+
+  static toFileName(jsFileName: string): string {
+    return jsFileName
+  }
+
+  static toFileAccess(jsFileRestricted: boolean): FileAccess {
+    return {
+      restricted: jsFileRestricted,
+      // TODO - Implement the rest of the properties when they are added to js-dataverse
+      latestVersionRestricted: false,
+      canBeRequested: false,
+      requested: false
+    }
+  }
+
+  static toFileType(jsFileContentType: string): FileType {
+    return new FileType(jsFileContentType)
+  }
+
+  static toFileSize(jsFileSize: number): FileSize {
+    return new FileSize(jsFileSize, FileSizeUnit.BYTES)
+  }
+
+  static toFileDate(
+    jsFileCreationDate?: Date,
+    jsFilePublicationDate?: Date,
+    jsFileEmbargo?: JSFileEmbargo
+  ): FileDate {
+    if (jsFilePublicationDate) {
+      if (jsFileEmbargo) {
+        return { type: FileDateType.METADATA_RELEASED, date: jsFilePublicationDate }
+      }
+      return { type: FileDateType.PUBLISHED, date: jsFilePublicationDate }
+    }
+    if (jsFileCreationDate) {
+      return { type: FileDateType.DEPOSITED, date: jsFileCreationDate }
+    }
+    throw new Error('File date not found')
+  }
+
+  static toFileDownloads(): number {
+    return 0 // This is always 0 because the downloads come from a different endpoint
+  }
+
+  static toFileLabels(jsFileCategories?: string[], jsFileTabularTags?: string[]): FileLabel[] {
+    const fileLabels: FileLabel[] = []
+    if (jsFileCategories) {
+      jsFileCategories.forEach((category) => {
+        fileLabels.push({ value: category, type: FileLabelType.CATEGORY })
+      })
+    }
+    if (jsFileTabularTags) {
+      jsFileTabularTags.forEach((tabularTag) => {
+        fileLabels.push({ value: tabularTag, type: FileLabelType.TAG })
+      })
+    }
+
+    // TODO - Add custom labels when they are added to js-dataverse
```

Comment:
Has any issue been created for this? Is this TODO up to date with the current state of development?

Reply:
We talked about this and we decided that custom labels are already retrieved as category tags so no TODO needed, I remove it